Services/AccessControl/classes/class.ilObjRoleGUI.php

```diff
@@ -609,6 +615,11 @@ public function saveObject()
 			include_once './Services/AccessControl/classes/class.ilObjRole.php';
 			$this->loadRoleProperties($this->role = new ilObjRole());
 			$this->role->create();
+			// cat-tms-patch start
+			$this->role->createTMSSettings();
+			$this->loadTMSRoleSettings($this->role);
+			$this->role->updateTMSSettings();
+			// cat-tms-patch end
```

Comment:
Gibt es einen Grund, warum dieser Block nicht in `ilObjRole::create` drin ist? IMO gehört er da rein. Dann müssen diese ganzen Methoden auch nicht `public` sein und Kapselung des Objekts ist besser bewahrt.

Reply:
Ja den Grund gibt es. Der nennt sich lokal Rolle. Es gibt keine Property die bestimmt ob es eine lokale oder globale Rolle ist. Wenn ich das in der ilObjRole::create packen, dann gibt es die Einträge auch für die lokalen Rollen. Und das wollen wir nicht.

Reply:
Ok, schade. Bündel dabb doch bitte diese 3 Schritte in einer Method und ruf die, z.B. `afterCreateGlobalRole` oder so.